### DISCUSSION TITLE
Different ways to have the grid view appear

### DISCUSSION THREAD

**Discussion by FilippoGuenzi** [dutchigor/json-grid-viewer#4] [Ideas]
The "open with" menu only appears when the folder the file is in is opened.
It would be nice if the "open with" menu could appear when just the file is opened.
It would be even better if the option was available in the command palette.
And it would be even better if this view could appear as a preview.

**Comment by dutchigor** [comment]
Hi Filippo,

Thanks for the good suggestions. I just took a quick look to see what is available at the moment and here's what I got so far:
* Since the file is not visible in the explorer panel when only the file is opened, there is no way to show the "Open With..." option there. However, if you right click on the tab name at the top, you have the option "Reopen Editor With...". This provides the same functionality.
* It is also possible to get the "open with" dialog from the command palette using the command "View: Reopen Editor With..."
* As for the preview, I purposely decided to go for the "open with" approach because I planned to allow for editing directly in the grid view. The "open with" approach is the default that VS code provides for custom editors. However, I have not really had the time to add this functionality and indeed I should onsider if I will add editing or otherwise add a preview button.
